dnf-behave-tests/dnf/vars-parsing.feature

```diff
@@ -0,0 +1,12 @@
+Feature: Correct parsing of dnf/vars directory
+ 
+ 
+@bz2091636
+Scenario: Using dnf with non-files in /etc/dnf/vars
+  Given I create directory "/{context.dnf.installroot}/etc/dnf/vars"
+    And I create directory "/{context.dnf.installroot}/tmp"
+  Given I successfully execute "ln -s {context.dnf.installroot}/tmp {context.dnf.installroot}/etc/dnf/vars/symlink"
+    And I use repository "dnf-ci-fedora"
+   When I execute dnf with args "repolist"
+   Then the exit code is 0
+    And stdout contains "repo id"
```

Comment:
This is a very weak check, what is the full expected output and could you check it in its entirety?

Reply:
I believe that the main part for the bug is test of the return code. The test of outputs is nice to have - it provides expected output after loading wars and even small peace is enough. And test that `stderr` does not contain `traceback` is also nice.

Reply:
Well I think that the test see a different output. There should be only `dnf-ci-fedora` repository because repolist is run together with installroot option. To test it just change expected RC to 1 and run it with patched RPM. The bug resulted in crash that prevent any output and it created a traceback, therefore the test doe not need to test exact match with output. In `repolist.feture` I've found following line `And stdout contains "dnf-ci-fedora\s+dnf-ci-fedora test repository\s+enabled"`. May be it can be used also for the test. What do you think?
PS: I guess that you did not push the latest changes.